src/main/scala/com/cognite/sdk/scala/v1/fdm/instances/InstanceDefinition.scala

```diff
@@ -311,6 +315,10 @@ object InstanceDefinition {
       t: PropertyType
   ): Either[DecodingFailure, InstancePropertyValue] =
     t match {
+      case PropertyType.EnumProperty(_, _) =>
+        Decoder[String]
+          .decodeJson(propValue)
+          .map(InstancePropertyValue.Enum.apply)
```

Comment:
and here we don't try to validate if value is matching one of possible values (or unknown value)

Reply:
We could, or we can rely on FDM to do it.

Reply:
might be good to catch early when writing and double-check when reading, but just trusting the service to validate in both cases is probably ok